### PR TITLE
Better crab-17 behavior and better code

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Better crab-17 behavior and better code
Crab-17 will teleport within its z-level instead of teleporting across z-levels (good at multi-z-level station)
and changed the code better

## Why It's Good For The Game
code cleaning
Also, it's a bit overwhelming in specific situations (low pop, multiple z station), so it will be less annoying in those situations.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

</details>

## Changelog
:cl:
code: touched crab-17 code a little better. It'll be logged too.
balance: crab-17 will teleport within the same z-level of it.
balance: crab-17 will be less annoying when it's spawned lower than 20 population. It's easy to kill and less teleport when there are less active players.
balance: crab-17 will do damage to itself until it has only 25 HP (self-damage won't let it teleporting). You still need to find it.
/:cl:

### CODE REVIEW

**Comment by CydiaLamiales** [comment]
Could you maybe make it self-destruct again? After 12 minutes maybe.

**Comment by EvilDragonfiend** [comment]
> Could you maybe make it self-destruct again? After 12 minutes perhaps .

I changed it better at low pop
but also, I don't want to self-destruction just hoping it'll be gone naturally, so I took your opinion partly. It will deal damage to intself until it has 25 HP, and a player will be needed to finish it.

**Comment by EvilDragonfiend** [comment]
also, the new commit is tested in my local too.

File: code/game/objects/items/crab17.dm [hunk 1]
```diff
@@ -195,8 +225,12 @@
 
 	// Oh no, it RUNS AWAY!!!
 	if(obj_integrity && obj_integrity < next_health_to_teleport) // checks if obj_integrity is positive first
-		next_health_to_teleport -= RUN_AWAY_THRESHOLD_HP
-		var/turf/targetturf = get_safe_random_station_turfs()
+		calculate_runaway_condition()
+		var/turf/targetturf
+		for(var/i in 1 to 100) // teleporting across z-levels is painful
+			targetturf = get_safe_random_station_turfs()
+			if(targetturf.z == src.z)
```
Comment (itsmeow) [inline code/game/objects/items/crab17.dm]: use `get_virtual_z_level`

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Comment by EvilDragonfiend** [comment]
addressed

File: code/game/objects/items/crab17.dm [hunk 2]
```diff
@@ -229,7 +229,7 @@
 		var/turf/targetturf
 		for(var/i in 1 to 100) // teleporting across z-levels is painful
 			targetturf = get_safe_random_station_turfs()
-			if(targetturf.z == src.z)
+			if(targetturf.z == src.get_virtual_z_level())
```
Comment (itsmeow) [inline code/game/objects/items/crab17.dm]: you need to do it for both zs

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/items/crab17.dm (hunk 2 above)
Comment (EvilDragonfiend) [reply code/game/objects/items/crab17.dm]: addressed

**Review by itsmeow — APPROVED** [review]

**Pull request merged**